src/containers/EventCreationPage.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useState, useEffect } from "react";
+import "../styles/EventCreationPage.css";
+import Typography from "@material-ui/core/Typography";
+import Button from "@material-ui/core/Button";
+import CancelModal from "../components/EventCreationPage/CancelModal";
+import Map from "../components/EventCreationPage/Map";
+import NextButton from "../components/EventCreationPage/NextButton";
+import BackButton from "../components/EventCreationPage/BackButton";
+import FormField from "../components/EventCreationPage/FormField";
+import Stepper from "../components/EventCreationPage/Stepper";
+import SelectDateModal from "../components/EventCreationPage/SelectDateModal";
+import { useMutation } from "react-apollo";
+import { ADD_EVENT } from "../graphql/mutations/templates/events";
+import { useEventMutation } from "../graphql/mutations/hooks/events";
+import { isObjectType } from "graphql";
+
+const EventCreationPage = () => {
+  const [openCancelModal, setOpenHandleModal] = useState(false);
+  const [openDateModal, setOpenDateModal] = useState(false);
+  const [complete, setComplete] = useState(false);
+
+  const [eventName, setEventName] = useState<string>("");
+  const [eventDate, setEventDate] = useState<Date | null>(null);
+  const [eventLocation, setEventLocation] = useState<string>("");
+
+  const [activeStep, setActiveStep] = useState<number>(0);
+
+  const handleOpenCancelModal = () => setOpenHandleModal(true);
+  const handleCloseCancelModal = () => setOpenHandleModal(false);
+
+  const handleOpenDateModal = () => setOpenDateModal(true);
+  const handleCloseDateModal = () => setOpenDateModal(false);
+
+  const handleNameChange = (e: any) => {
+    setEventName(e.target.value);
+  };
+  const handleDateChange = (e: any) => {
+    setEventDate(e.target.value);
+  };
+  const handleLocationChange = (e: any) => {
+    setEventLocation(e.target.value);
+  };
+
+  const handleNext = () => {
+    setActiveStep((prevStep) => prevStep + 1);
+  };
+  const handleBack = () => {
+    setActiveStep((prevStep) => prevStep - 1);
+  };
+
+  useEventMutation(
+    {
+      name: eventName,
+      eventDate: eventDate || new Date(),
+      createdBy: 1,
+      isActive: true,
+    },
+    complete,
+    setComplete
+  );
+
+  const handleComplete = () => {
+    setComplete(true);
+    window.location.href = "/events/";
```

Comment:
can we try using react-router's `useHistory` hook instead? https://stackoverflow.com/questions/31079081/programmatically-navigate-using-react-router
It should make it work more like a single page application (won't refresh the page).